Ex2 js basics part2

ruminmaxim opened the pull request.

ruminmaxim committed:
Function which detect type of variable

ruminmaxim committed:
Function which shows off a content of an array and its length

ruminmaxim committed:
Function which count amount of zeros, odds and evens

ruminmaxim committed:
Function which checks duplicates in an array and return boolean result ("true" if all elements are same, else "false")

ruminmaxim committed:
Function which return the biggest element in an array

aneelia commented on src/ex2_js-basics-part2/task-01.js:
```diff
@@ -0,0 +1,14 @@
+function typeOfVar(x) {
+var result;
+var y = typeof x;
+if (y === 'number') {
```
в целом, верно, но можно упростить-слишком много дублирования кода. подумай, как. если не получится, на занятии объясню
и еще, есть такое интересное значение - NaN, которое означает "не число". Но тип у него - number. Надо бы проверку добавить

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,23 @@
+function arrCounter(arr) {
+var countZero = 0;
+var countOdd = 0;
+var countEven = 0
```
можно три раза не писать var, просто через запятую объявить переменные и присвоить им значение

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,23 @@
+            } 
+    }
+ var final = [countEven, countOdd, countZero];
+ console.log (final);
```
консоль убрать, и переменная final  не нужна. просто return [countEven, countOdd, countZero];

aneelia commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,12 @@
+function duplicatesCheck(arr) {
+
+var buffer = arr[0];
+var check = true;
```
можно просто объявить переменную, ты ее определяешь ниже

aneelia commented on src/ex2_js-basics-part2/task-05.js:
```diff
@@ -0,0 +1,12 @@
+function biggestNumber(arr) {
+var buffer = arr[0];
+for (var i=0; i<arr.length; i++) {
+if (arr[i] <= buffer) continue;
```
смысла вообще никакого в этой строке нет

ruminmaxim committed:
1. Function which define type of variable is less now. "NaN" Check is added too. (file task-01.js)
2. Variables in Counting Function are listed by shorter way. Useless strings are deleted (file task-03.js)
3. Variables are listed by shorter way in function, which checks dublicates (file task-04.js)
4. Useless strins is deleted in function which defines the biggest number (file task-05.js)
